source/server-discovery-and-monitoring/tests/integration/closeInUse-pool-clear.yml

```diff
@@ -0,0 +1,108 @@
+# Test connection pool clear(closeInUse:true).
+runOn:
+    # This test requires failCommand appName, and
+    # failCommand blockConnection with closeConnection:true (SERVER-53512).
+    - minServerVersion: "4.9"
+
+database_name: &database_name "sdam-tests"
+collection_name: &collection_name "closeInUse"
+
+data: []
+
+tests:
+  - description: pool clear close in use true generates expected events
+    useMultipleMongoses: false,
+    clientOptions:
+      connectTimeoutMS: 500, 
+      heartbeatFrequencyMS: 500
+      appname: "closeInUse"
+      retryReads: false
+      minPoolSize: 0
+    operations:
+      - name: insertMany
+        object: collection
+        arguments:
+          documents:
+            - _id: 1
+      - name: startThread
+        object: testRunner
+        arguments:
+          name: thread1
+      # simulate a long-running query
+      - name: runOnThread
+        object: testRunner
+        arguments:
+          name: thread1
+          operation:
+            name: find
+            object: collection
+            arguments:
+              filter: { "$where": "sleep(5000) || true" }
+      - name: configureFailPoint
+        object: testRunner
+        arguments:
+          failPoint:
+            configureFailPoint: failCommand
+            mode: { times: 1 }
+            data:
+                failCommands: ["hello"]
+                blockConnection: true
+                blockTimeMS: 1000
+                appName: closeInUse
+      - name: waitForEvent
+        object: testRunner
+        arguments:
+          event: ConnectionClosed
+          count: 1
+      - name: waitForEvent
+        object: testRunner
+        arguments:
+          event: PoolClearedEvent
+          count: 1
+      - name: waitForEvent
```

Comment:
Can we change the `waitForEvent` orderings so it goes:
1. PoolClearedEvent
2. ConnectionCheckedInEvent
3. ConnectionClosedEvent

This doesn't change the test in a meaningful way, but it makes it easier to read as this is the order the events actually occur in.

Also can we add the following comments:

- To the waitForEvent with `PoolClearedEvent`:
```
# After timing out, the monitor should clear the pool with closeInUseConnections=true
```

- To the waitForEvent with `ConnectionCheckedInEvent`:
```
# Wait until the connection used for the find is returned to the pool and subsequently closed.
```